pandora-client-web/src/ui/screens/spaceConfiguration/spaceConfiguration.tsx

```diff
@@ -273,31 +275,30 @@ export function SpaceConfiguration({ creation = false }: { creation?: boolean; }
 				{ configurableElements }
 				<div className='input-container'>
 					<label>Features (cannot be changed after creation)</label>
-					<ul>
-						{
-							SPACE_FEATURES.map((feature) => (
-								<li key={ feature.id }>
-									<input type='checkbox'
-										id={ `${idPrefix}-feature-${feature.id}` }
-										checked={ currentConfig.features.includes(feature.id) }
-										onChange={ (event) => {
-											if (event.target.checked) {
-												if (!currentConfig.features.includes(feature.id)) {
-													setModifiedData({ features: [...currentConfig.features, feature.id] });
-												}
-											} else {
-												setModifiedData({ features: currentConfig.features.filter((f) => f !== feature.id) });
+					{
+						SPACE_FEATURES.map((feature) => (
+							(feature.id !== 'development' || (feature.id === 'development' && isDeveloper)) &&
+							<div key={ feature.id }>
+								<input type='checkbox'
+									id={ `${idPrefix}-feature-${feature.id}` }
+									checked={ currentConfig.features.includes(feature.id) }
+									onChange={ (event) => {
+										if (event.target.checked) {
+											if (!currentConfig.features.includes(feature.id)) {
+												setModifiedData({ features: [...currentConfig.features, feature.id] });
 											}
-										} }
-									/>
-									<label htmlFor={ `${idPrefix}-feature-${feature.id}` }>{ feature.name }</label>
-								</li>
-							))
-						}
-					</ul>
+										} else {
+											setModifiedData({ features: currentConfig.features.filter((f) => f !== feature.id) });
+										}
+									} }
+								/>
+								<label htmlFor={ `${idPrefix}-feature-${feature.id}` }>{ ' ' + feature.name }</label>
+							</div>
+						))
+					}
 				</div>
 				{
-					currentConfig.features.includes('development') &&
+					currentConfig.features.includes('development') && isDeveloper &&
```

Comment:
should we also enforce this in server side too, since this only hides the feature and do not prevent anyone from creating a room with development features if they know how

Reply:
Yes, I wrote so in my comment in the PR description~ *giggles*

Reply:
you can add enforcment by checking the feature and the role of the creator, something like this:

in: `pandora-server-directory/src/spaces/spaceManager.ts`

```diff
-       public async createSpace(config: SpaceDirectoryConfig, owners: AccountId[]): Promise<Space | 'failed' | 'spaceOwnershipLimitReached'> {
+       public async createSpace(config: SpaceDirectoryConfig, owners: AccountId[], creator: Account): Promise<Space | 'failed' | 'spaceOwnershipLimitReached'> {
                Assert(owners.length > 0, 'Space must be created with some owners');
 
+               if (config.features.includes('development') && !creator.roles.isAuthorized('developer')) {
+                       return 'failed';
+               }
```